### DISCUSSION TITLE
Is it a good idea/bad idea to edit changelogs later on?

### DISCUSSION THREAD

**Discussion by HonkingGoose** [semantic-release/semantic-release#2080] [Q&A] [answered]
## Introduction

The Renovate project is thinking of editing their major version GitHub release notes so they have a section on _why_ we're making certain changes, and to _give some background information_ if needed. Right now our release notes don't always properly explain the reason behind some changes.

Example of one of the release notes that we're thinking of editing: https://github.com/renovatebot/renovate/releases/tag/26.0.0

## My own research

Is editing the GitHub release notes later on a good/bad thing to do? I've read the discussions in this repo, and checked the frequently asked questions, but did not find anything that directly answers my question.

I see that it's possible to use the commit body to add extra messages that will show up in the generated release notes, but this relies on getting those commit messages right the first time, I think?

I understand that the philosophy of `semantic-release` is "kill all humans" and "no sentimental releases", and maybe editing the GitHub release notes later on is a stupid thing to do? 😄 

## Questions

In short, the questions that I have are:

- If we edit the GitHub release notes, does `semantic-release` clobber over them in a newer run?
- Is editing the release notes after they're generated even a good idea to begin with?
- Maybe you maintainers have a better idea/process/plan?

**Comment by gr2m** [comment, marked as answer]
> * If we edit the GitHub release notes, does `semantic-release` clobber over them in a newer run?

No

> * Is editing the release notes after they're generated even a good idea to begin with?

I edit release notes myself all the time. For example, I squash & commit all my pull requests, but some PRs add more than one feature / bug fix. In that case I go in and edit the release notes manually. I also edited most of the breaking releases as you plan to do with https://github.com/renovatebot/renovate/releases/tag/26.0.0, to add notes on how to migrate. Breaking release notes are the ones being most read by humans, so totally worth investing in them.

> * Maybe you maintainers have a better idea/process/plan?

I wanted to build a plugin for a while which permits pull request authors / maintainers to author hand-written release notes and using that as an additional / alternative to reading out the commit notes. That'd be a really nice flow for GitHub I think, but no idea when/if I get to work on that 🤣

**Reply by HonkingGoose** [reply]
Thank you for answering my questions!

**Reply by travi** [reply]
i second this advice. another use case where i suggest this to my teams somewhat often is when a commit is merged with an incorrect type, resulting in something that was intended to trigger a release, but ends up getting missed. education about the commit convention keeps occurrence of this low, but it does happen. 

once another commit is merged that triggers a release, it is worth adding the missed feature into the release notes so that the release notes are still complete and dont leave out the feature that was missed by the automated release process.

**Reply by HonkingGoose** [reply]
Thanks for the advice to edit the release notes to add a missed feature, in case the conventional commit was wrong!
